### DISCUSSION TITLE
Is there a workaround?

### DISCUSSION THREAD

**Discussion by wiverson** [arthelokyo/astrowind#17] [Q&A]
https://github.com/withastro/astro/issues/4536

Just want to be able to automatically pull in the images from frontmatter to display on the home page. Is there a workaround you might suggest?

**Comment by prototypa** [comment]
Hi @wiverson 

One important thing that I'm not sure is if you are using the **Picture** component from [@astrojs/image](https://docs.astro.build/en/guides/integrations-guide/image/) or the one that is implemented in AstroWind (https://github.com/onwidget/astrowind/blob/main/src/components/core/Picture.astro).

Please confirm that for further investigation.

PD: In the case of the issue you refer to (https://github.com/withastro/astro/issues/4536), I think the problem is directly related to Astro or Vite and the loading of images in relative paths or in the `public` folder

**Reply by wiverson** [reply]
I was using the astrojs/image, I'm trying the theme's Picture.astro now. Crossed fingers. :)

**Reply by prototypa** [reply]
@wiverson , please let me know if it worked. If not, give me more details to help you.

Also note that if you are using an alias on the posts in Markdown(ex: images: ~/assets/..), to be able to import that I think you must have aliases configured in Vite:

```javascript
{
  vite: {
    resolve: {
      alias: {
        "~": path.resolve(__dirname, "./src"),
      },
    },
  }
}
```

https://github.com/onwidget/astrowind/blob/main/astro.config.mjs#L32

**Reply by wiverson** [reply]
Ok, I got it working (yay!) but I am still confused by how all of the path resolution stuff works. FWIW this is good practice for an old Java dev playing around in JS/TS world. :)

Here is an updated / hacky WIP that works.

https://gist.github.com/wiverson/a2979d17b5d3cf0d0908dede36ff4114

Interestingly, it works, but VS Code is giving me an error on this line:

https://gist.github.com/wiverson/a2979d17b5d3cf0d0908dede36ff4114#file-highlightedposts-astro-L82

...for the hard-coded version of the string, but not complaining about the version in the loop:

https://gist.github.com/wiverson/a2979d17b5d3cf0d0908dede36ff4114#file-highlightedposts-astro-L68

<img width="874" alt="image" src="https://user-images.githubusercontent.com/873850/187516664-7c1036c7-d1cf-42c7-ac43-ba34910f9b6a.png">

**Reply by wiverson** [reply]
> @wiverson , please let me know if it worked. If not, give me more details to help you.
> 
> Also note that if you are using an alias on the posts in Markdown(ex: images: ~/assets/..), to be able to import that I think you must have aliases configured in Vite:
> 
> ```js
> {
>   vite: {
>     resolve: {
>       alias: {
>         "~": path.resolve(__dirname, "./src"),
>       },
>     },
>   }
> }
> ```
> 
> https://github.com/onwidget/astrowind/blob/main/astro.config.mjs#L32

This is set up correctly.

**Reply by prototypa** [reply]
You can review the updates that I have been making to the template, perhaps they will help you achieve your goals.

For Ex: [HighlightedPosts.astro](https://github.com/onwidget/astrowind/blob/main/src/components/widgets/HighlightedPosts.astro)

The error that is shown (although it works) is related to typescript, they must have the type assigned incorrectly because according to the documentation the src attribute also allows the import function to be used directly.

Note: I'm still not satisfied with the final result of the template (I'm still understanding Astro), that's why it's still in version 0.2.0

**Reply by wiverson** [reply]
Ok, it's working now, here's my version:

https://gist.github.com/wiverson/697779257b9979ca07f3048f2bcad97d

...looks like it shares a bit in common with yours. Out of curiosity, why do you have your own Picture component?

**Reply by prototypa** [reply]
Hi @wiverson. Great that it works now.

I have my own Picture component because when I first started using it I didn't have a way to add a Tailwind CSS style class specifically to the child <img> tag that is rendered. (And also because when I started with this I knew little about how Astro worked)

I'm going to review these days how @astrojs/image has been updated and maybe I'll transform all that.

**Reply by wiverson** [reply]
Well, I'm learning a ton every day. Between Astro 1.0 and SvelteKit changes, front-end is pretty wild, lol.

Looks like the Tailwind classes I'm setting on Image are working, I imagine Picture has a similar passthrough...?

**Reply by prototypa** [reply]
Yeah!!, front-end is pretty wild 😄 

Related to **Images** and **Pictures** it's a super silly detail (check the following code). Soon I will refactor all  that image stuff in the template and I will surely remove the custom **Picture** component.

```html
<Image {attrs...} class="tw-shadow-md" />

<!-- renders to -->

<img {attrs...} class="tw-shadow-md />
```

But: (there are no ways to apply some styles directly to the image)
```html
<Picture {attrs...} class="tw-shadow-md" />

<!-- renders to -->

<picture {attrs...} class="tw-shadow-md">
  <source {...} />
  <img {...image} {loading} {decoding} {alt} />
</picture>
```
